pkg/endpoint/endpoint.go

```diff
@@ -197,36 +203,53 @@ func RunK8sCiliumEndpointSyncGC() {
 					}
 				}
 
-				clusterPodSet := map[string]bool{}
-				clusterPods, err := k8sClient.CoreV1().Pods("").List(meta_v1.ListOptions{})
-				if err != nil {
-					return err
-				}
-				for _, pod := range clusterPods.Items {
-					podFullName := pod.Name + ":" + pod.Namespace
-					clusterPodSet[podFullName] = true
-				}
+				var (
+					listOpts = meta_v1.ListOptions{Limit: 10}
+					loopStop = time.Now().Add(CiliumEndpointGCInterval)
+				)
 
-				// "" is all-namespaces
-				ceps, err := ciliumClient.CiliumEndpoints(meta_v1.NamespaceAll).List(meta_v1.ListOptions{})
-				if err != nil {
-					scopedLog.WithError(err).Debug("Cannot list CEPs")
-					return err
-				}
-				for _, cep := range ceps.Items {
-					cepFullName := cep.Name + ":" + cep.Namespace
-					if _, found := clusterPodSet[cepFullName]; !found {
-						// delete
-						scopedLog = scopedLog.WithFields(logrus.Fields{
-							logfields.EndpointID: cep.Status.ID,
-							logfields.K8sPodName: cepFullName,
-						})
-						scopedLog.Debug("Orphaned CiliumEndpoint is being garbage collected")
-						if err := ciliumClient.CiliumEndpoints(cep.Namespace).Delete(cep.Name, &meta_v1.DeleteOptions{}); err != nil {
-							scopedLog.WithError(err).Debug("Unable to delete CEP")
+			perCEP:
+				for time.Now().Before(loopStop) { // stop the loop if it takes too long
+					ceps, err := ciliumClient.CiliumEndpoints(meta_v1.NamespaceAll).List(listOpts)
+					switch {
+					case ceps.Continue != "" && err != nil:
+						// this is ok, it means we saw a 410 ResourceExpired error but we
```

Comment:
The comment is helpful, but is this documented somewhere? I'd suggest putting a link in a comment here to said documentation.

Reply:
Good point. I was, in fact, checking this incorrectly. I've fixed it.